pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -116,11 +135,13 @@ func (d *DualWriterMode2) Get(ctx context.Context, name string, options *metav1.
 		log.Info("object from legacy and storage are not equal")
 	}
 
-	// if there is no object in storage, we return the object from legacy
-	if objStorage == nil {
-		return objLegacy, nil
+	if objStorage != nil {
+		if err := updateRVOnLegacyObj(objStorage, objLegacy); err != nil {
+			log.WithValues("storageObject", objStorage, "legacyObject", objLegacy).Error(err, "could not update resource version")
+		}
 	}
```

Comment:
I don't understand that part 🤔 . Why do we need to ensure the RV is coming from Unified ?

Reply:
Thinking about it ... overwise the concurrency check on update will fail. Good thinking.

Reply:
I might be wrong but I dont think we need to do the same on list as it's only `get` which is called before an update to check if the object exists and from what I remember it's this `get` method which is being called. So i think we only need to enforce this here?

Reply:
Possibly !